airflow/providers/google/cloud/transfers/azure_fileshare_to_gcs.py

```diff
@@ -0,0 +1,217 @@
+                '"directory", define a path that ends with a slash "/" or '
+                'leave it empty for the root of the bucket.'
+            )
+            raise AirflowException(
+                'The destination Google Cloud Storage path ' 'must end with a slash "/" or be empty.'
+            )
+
+    def execute(self, context):
+        azure_fileshare_hook = AzureFileShareHook(self.wasb_conn_id)
+        files = azure_fileshare_hook.list_files(
+            share_name=self.share_name, directory_name=self.directory_name
+        )
+
+        gcs_hook = GCSHook(
+            google_cloud_storage_conn_id=self.gcp_conn_id,
+            delegate_to=self.delegate_to,
+            impersonation_chain=self.google_impersonation_chain,
+        )
+
+        # pylint: disable=too-many-nested-blocks
+        if not self.replace:
+            # if we are not replacing -> list all files in the GCS bucket
+            # and only keep those files which are present in
+            # S3 and not in Google Cloud Storage
+            bucket_name, object_prefix = _parse_gcs_url(self.dest_gcs)
+            existing_files_prefixed = gcs_hook.list(bucket_name, prefix=object_prefix)
+
+            existing_files = []
+
+            if existing_files_prefixed:
+                # Remove the object prefix itself, an empty directory was found
+                if object_prefix in existing_files_prefixed:
+                    existing_files_prefixed.remove(object_prefix)
+
+                # Remove the object prefix from all object string paths
+                for f in existing_files_prefixed:
+                    if f.startswith(object_prefix):
+                        existing_files.append(f[len(object_prefix) :])
+                    else:
+                        existing_files.append(f)
+
+            files = list(set(files) - set(existing_files))
+            if len(files) > 0:
+                self.log.info('%s files are going to be synced: %s.', len(files), files)
+            else:
+                self.log.info('There are no new files to sync. Have a nice day!')
+
+            if files:
+                for file in files:
+                    with NamedTemporaryFile() as f:
+                        azure_fileshare_hook.get_file_to_stream(
+                            stream=f,
+                            share_name=self.share_name,
+                            directory_name=self.directory_name,
+                            file_name=file,
+                        )
+                        f.flush()
+
+                        # There will always be a '/' before file because it is
+                        # enforced at instantiation time
+                        dest_gcs_object = object_prefix + file
+                        gcs_hook.upload(bucket_name, dest_gcs_object, f.name, gzip=self.gzip)
+
+            self.log.info("All done, uploaded %d files to Google Cloud Storage", len(files))
```

Comment:
Due to identation, this part will execute only when `self.replace` is False.

Reply:
oops I messed identation a bit here, fixed